### ISSUE TITLE
[Report] Unable to connect to Wifi

### ISSUE THREAD

**Issue by iMonZ** [holoiso-eol/holoiso#466]
## Your Issue:
```
*type your problem here*
```

cannot connect to my WPA3 Wifi

## Your device's full specs
```
```
Lenovo P15 g2

## Your OS version (`cat /etc/os-release | grep VARIANT_ID | cut -d '"' -f 2`):
```
snapshot19700101_1011-beta2a
```

## (Only for SteamOS Session loading issues) Have you tried launching `gamescope-session` from terminal? If yes, provide a log.
```
yeah pubg crash bhai
```

## Your installation type:
```
deckperience
```

## Have you updated your installation recently?
```
nawww
```

So on my official steamdeck inside plasma wpa3 worked perfectly fine even on the zeroday version (so the first SteamOS version since I had one of the first steamdecks) on my laptop i get this error:
![41A4D64F-2368-4CE5-BE56-10C2B1F2683A](https://user-images.githubusercontent.com/11319871/185758749-68997e64-9ce7-42db-9a89-91f78d81992e.jpeg)

**Comment by jrcharney** [comment]
Which version are you using when you start up? `linux-neptune` or `linux-holoiso`?

**Comment by iMonZ** [comment]
> Which version are you using when you start up? `linux-neptune` or `linux-holoiso`?

I think the default.
What’s the difference?

**Comment by irfanhakim-as** [comment]
> > Which version are you using when you start up? `linux-neptune` or `linux-holoiso`?
> 
> I think the default. What’s the difference?

I don't remember what the default is but my guess it was `linux-neptune`. Your best bet is to use `linux-holoiso`. To my understanding, the former comes with drivers specific to only the Steam Deck, while the latter comes with more drivers to support most other devices/desktops like a typical Linux distro would. This might offer better compatibility for your device.

**Comment by jrcharney** [comment]
> > > Which version are you using when you start up? `linux-neptune` or `linux-holoiso`?
> > 
> > 
> > I think the default. What’s the difference?
> 
> I don't remember what the default is but my guess it was `linux-neptune`. Your best bet is to use `linux-holoiso`. To my understanding, the former comes with drivers specific to only the Steam Deck, while the latter comes with more drivers to support most other devices/desktops like a typical Linux distro would. This might offer better compatibility for your device.

More than likely. I've noticed that WiFi and Bluetooth work when `linux-holoiso` is selected. When it starts up in `linux-neptune`, nothing seems to work.

I'm actually interested in modifying the GRUB config file to use `linux-holo` by default.  It would be nice if there was a tutorial to edit it and to make sure that it doesn't get reset whenever a new kernel update happens.

**Comment by irfanhakim-as** [comment]
> > > > Which version are you using when you start up? `linux-neptune` or `linux-holoiso`?
> > > 
> > > 
> > > I think the default. What’s the difference?
> > 
> > 
> > I don't remember what the default is but my guess it was `linux-neptune`. Your best bet is to use `linux-holoiso`. To my understanding, the former comes with drivers specific to only the Steam Deck, while the latter comes with more drivers to support most other devices/desktops like a typical Linux distro would. This might offer better compatibility for your device.
> 
> More than likely. I've noticed that WiFi and Bluetooth work when `linux-holoiso` is selected. When it starts up in `linux-neptune`, nothing seems to work.
> 
> I'm actually interested in modifying the GRUB config file to use `linux-holo` by default. It would be nice if there was a tutorial to edit it and to make sure that it doesn't get reset whenever a new kernel update happens.

Then we're on the same boat haha but I still am unsure what the "right" method is to do that yet, esp for people w custom PC/rigs n not a popular handheld like the OXPs and the Aya Neos. See my issue on grub here #458

**Comment by jrcharney** [comment]
> Then we're on the same boat haha but I still am unsure what the "right" method is to do that yet, esp for people w custom PC/rigs n not a popular handheld like the OXPs and the Aya Neos. See my issue on grub here #458

Well, the good news is that the file was written quite thorough, it's just a matter of recognizing which lines need to be swapped and to put them in an order that if things go bad, you can go back and fix it.  I'm just not going to do it at 1AM when I'm just checking my email before going to bed.

It is something that I have written down on a sticky note and a card on my Kanban board.  I'm hoping to provide a good answer tomorrow especially since I should get it done by the end of the week when I plan on presenting my steam box to some friends who would be interested in this project.

I'm hoping I can provide an answer very soon.

**Comment by iMonZ** [comment]
![C106E9B3-3EE6-4FC8-B59F-093DA2429AD0](https://user-images.githubusercontent.com/11319871/186395204-53efec68-45dc-4fd1-a508-f8984c58727f.jpeg)

booting holoiso doesn't work for me...
The only thing that doesn't work in neptune is wifi and the speaker. I have researched and can't find anything to fix the speaker problem

**Comment by Hoshino324** [comment]
> > Then we're on the same boat haha but I still am unsure what the "right" method is to do that yet, esp for people w custom PC/rigs n not a popular handheld like the OXPs and the Aya Neos. See my issue on grub here #458
> 
> Well, the good news is that the file was written quite thorough, it's just a matter of recognizing which lines need to be swapped and to put them in an order that if things go bad, you can go back and fix it. I'm just not going to do it at 1AM when I'm just checking my email before going to bed.
> 
> It is something that I have written down on a sticky note and a card on my Kanban board. I'm hoping to provide a good answer tomorrow especially since I should get it done by the end of the week when I plan on presenting my steam box to some friends who would be interested in this project.
> 
> I'm hoping I can provide an answer very soon.

On the same boat, too. I am using an Ayaneo 2021 and trying to figure out how to change the default bootup to holoiso. Just don't know why it is so hard to change the grub and it is really hard for me to read through the codes in the grub document. Hope to see your answer soon. Orz

**Comment by jrcharney** [comment]
> booting holoiso doesn't work for me... The only thing that doesn't work in neptune is wifi and the speaker. I have researched and can't find anything to fix the speaker problem

Reading some of the other issues here, and I'm thinking of probably dealing with the easy part first, which may be to fix the sound.

A few thoughts:
1. What does `aplay -L` look like?
2. What does `aplay -l` look like? (This time with a lowercase L)
3. What does `cat /proc/asound/cards` look like?

If the wifi is not working after using `linux-holoiso` as the Linux kernel instead of `linux-neptune`, this could be more of an Apple hardware issue than a Linux distro issue. And from what I'm reading, about the Apple A15 Bionic processor, this thing shouldn't even exist inside of a laptop....unless you are running this off a iPhone hooked up to some kind of lapdock.

Could you post what `neofetch` says is in your computer?

**Comment by iMonZ** [comment]
Bootet with Neptune:

![image](https://user-images.githubusercontent.com/11319871/186457468-3a35e07e-aba7-42d5-8a27-af5fb514ffda.jpg)

![image](https://user-images.githubusercontent.com/11319871/186457580-6bc63e9d-76bf-419e-873e-f7a69914b18b.jpg)

![image](https://user-images.githubusercontent.com/11319871/186457705-819bf8c6-80f4-4d78-a7c9-940d91b09d4a.jpg)

![image](https://user-images.githubusercontent.com/11319871/186457892-ce7d0092-abca-4682-8db8-c5c12437b05c.jpg)


Here you go :)
Thanks for your help

**Comment by irfanhakim-as** [comment]
> If the wifi is not working after using `linux-holoiso` as the Linux kernel instead of `linux-neptune`, this could be more of an Apple hardware issue than a Linux distro issue. And from what I'm reading, about the Apple A15 Bionic processor, this thing shouldn't even exist inside of a laptop....unless you are running this off a iPhone hooked up to some kind of lapdock.
> 
> Could you post what `neofetch` says is in your computer?

It's a little confusing but the Apple A15 isn't really part of their computer specs, that's just a placeholder when creating a new issue on this repo that they hadn't removed and replaced with their own specs. Their actual device is the "Lenovo P15 g2" stated below the code block. regardless, seems the issue has been resolved. if it has, @iMonZ u can close the issue.

**Comment by iMonZ** [comment]
> > If the wifi is not working after using `linux-holoiso` as the Linux kernel instead of `linux-neptune`, this could be more of an Apple hardware issue than a Linux distro issue. And from what I'm reading, about the Apple A15 Bionic processor, this thing shouldn't even exist inside of a laptop....unless you are running this off a iPhone hooked up to some kind of lapdock.
> > Could you post what `neofetch` says is in your computer?
> 
> It's a little confusing but the Apple A15 isn't really part of their computer specs, that's just a placeholder when creating a new issue on this repo that they hadn't removed and replaced with their own specs. Their actual device is the "Lenovo P15 g2" stated below the code block. regardless, seems the issue has been resolved. if it has, @iMonZ u can close the issue.

Oh sorry didn’t see the placeholder.
So audio doesn’t work at all :(
And wifi is an kde issue as I see

**Comment by irfanhakim-as** [comment]
> > booting holoiso doesn't work for me... The only thing that doesn't work in neptune is wifi and the speaker. I have researched and can't find anything to fix the speaker problem
> 
> Reading some of the other issues here, and I'm thinking of probably dealing with the easy part first, which may be to fix the sound.
> 
> A few thoughts:
> 
>     1. What does `aplay -L` look like?
> 
>     2. What does `aplay -l` look like? (This time with a lowercase L)
> 
>     3. What does `cat /proc/asound/cards` look like?
> 
> 
> If the wifi is not working after using `linux-holoiso` as the Linux kernel instead of `linux-neptune`, this could be more of an Apple hardware issue than a Linux distro issue. And from what I'm reading, about the Apple A15 Bionic processor, this thing shouldn't even exist inside of a laptop....unless you are running this off a iPhone hooked up to some kind of lapdock.
> 
> Could you post what `neofetch` says is in your computer?

@iMonZ maybe u can try checking this out. Running `lspci -k` might help too in getting more details on ur device/audio drivers.

**Comment by jrcharney** [comment]
> > If the wifi is not working after using `linux-holoiso` as the Linux kernel instead of `linux-neptune`, this could be more of an Apple hardware issue than a Linux distro issue. And from what I'm reading, about the Apple A15 Bionic processor, this thing shouldn't even exist inside of a laptop....unless you are running this off a iPhone hooked up to some kind of lapdock.
> > Could you post what `neofetch` says is in your computer?
> 
> It's a little confusing but the Apple A15 isn't really part of their computer specs, that's just a placeholder when creating a new issue on this repo that they hadn't removed and replaced with their own specs. Their actual device is the "Lenovo P15 g2" stated below the code block. regardless, seems the issue has been resolved. if it has, @iMonZ u can close the issue.

It would have been easier to identify the problem if the placeholder values weren't use.

Since this is an Intel setup (which as someone using an AMD device, shouldn't really be too different except for some basic drivers that are set during the OS install) rather than an Apple, the sound should work.

Another thing to note, if you are using an Nvidia device for your GPU, you should be using the proprietary drivers. As much as we'd love to have working open source Nvidia drivers, you're better off with the proprietary stuff at the moment.

I seriously think the WiFi issue can be resolved by using `linux-holoiso` instead of `linux-neptune` at startup. You need to change to that kernel when GRUB shows up. It should be in the Advanced SteamOS options.

**Comment by iMonZ** [comment]
> > > If the wifi is not working after using `linux-holoiso` as the Linux kernel instead of `linux-neptune`, this could be more of an Apple hardware issue than a Linux distro issue. And from what I'm reading, about the Apple A15 Bionic processor, this thing shouldn't even exist inside of a laptop....unless you are running this off a iPhone hooked up to some kind of lapdock.
> > > Could you post what `neofetch` says is in your computer?
> > 
> > 
> > It's a little confusing but the Apple A15 isn't really part of their computer specs, that's just a placeholder when creating a new issue on this repo that they hadn't removed and replaced with their own specs. Their actual device is the "Lenovo P15 g2" stated below the code block. regardless, seems the issue has been resolved. if it has, @iMonZ u can close the issue.
> 
> It would have been easier to identify the problem if the placeholder values weren't use.
> 
> Since this is an Intel setup (which as someone using an AMD device, shouldn't really be too different except for some basic drivers that are set during the OS install) rather than an Apple, the sound should work.
> 
> Another thing to note, if you are using an Nvidia device for your GPU, you should be using the proprietary drivers. As much as we'd love to have working open source Nvidia drivers, you're better off with the proprietary stuff at the moment.
> 
> I seriously think the WiFi issue can be resolved by using `linux-holoiso` instead of `linux-neptune` at startup. You need to change to that kernel when GRUB shows up. It should be in the Advanced SteamOS options.

But holoiso doesn’t boot at all that’s the problem. And how can I disable the NVIDIA drivers for upgrading my drivers?

**Comment by Hoshino324** [comment]
Well, I have changed all the device names in the grub document to my device with boot up option of holoiso, but it still doesn't work and tries to startup with Neptune. Definitely no ideas about how to change the default setting.

**Comment by jrcharney** [comment]
> Well, I have changed all the device names in the grub document to my device with boot up option of holoiso, but it still doesn't work and tries to startup with Neptune. Definitely no ideas about how to change the default setting.

Well, obviously. Changing the names of the devices would have no effect. On the bright side, at least we know that you can rename them.

Hopefully you didn't edit the `/boot/grub/grub.cfg` file which gets autogenerated and has an important message at the top?  You know, this one.

```
#
# DO NOT EDIT THIS FILE
# 
# It is automatically generated by grub-mkconfig using templates
# from /etc/grub.d and settings from /etc/default/grub
# 
```

I think `/etc/default/grub` might be a good file for guidance.  And for all you Aya Neo fans, I think you'll definitely find line 9 relevant. :laughing:

**Comment by jrcharney** [comment]
> On the same boat, too. I am using an Ayaneo 2021 and trying to figure out how to change the default bootup to holoiso. Just don't know why it is so hard to change the grub and it is really hard for me to read through the codes in the grub document. Hope to see your answer soon. Orz

I got some great news. After almost ruining my `/etc/default/grub` file because `grub-customizer` is litterally Linux Malware, I was able to get a working copy of the `/etc/default/grub` file and was able to successfully swap `linux-neptune` with `linux-holoiso`. The "how" I was able to do it I haven't figured out because I was sure that `grub-customizer` didn't save my changes (which apparently it did) and didn't say where those changes were made (probably `/etc/grub.d/10_linux_proxy`)?  It's also in some proxy script.

TLDR: Everything works at start up. No tomfoolery trying to get into the Advanced menu. No problems with the Wifi or bluetooth. It just works!

I still need to figure out what's up with the `os-prober` and why it won't look for my other drive with the Windows partition on it which would be better than using a virtual machine.  There needs to be some better, more friendly software to work with to fix Grub issues, and I promise I will find out, especially since I have another computer that I will need to repeat this process on to confirm my results.
